factory.py

```diff
@@ -23,17 +31,28 @@ def get_factories(task: Union[str, FactoryTypes, Tasks]) -> List["Factory"]:
         List[Factory]: List of factory classes.
     """
 
-    task: FactoryTypes = FactoryTypes(task)
-    if task == FactoryTypes.TEXT_GENERATION:
-        # text-generation
-        from text_generation.main import Factory as TextGenerationFactory
-
-        cls = [TextGenerationFactory]
-    elif task == FactoryTypes.VISION:
-        # vision
-        from vision.factory import CompressionFactory as VisionFactory
+    if not job_service or job_service == JobServices.KOMPRESS:
+        task: FactoryTypes = FactoryTypes(task)
 
-        cls = [VisionFactory]
+        # Kompress
+        if task == FactoryTypes.TEXT_GENERATION:
+            # text-generation
+            from text_generation.main import Factory as TextGenerationFactory
+
+            cls = [TextGenerationFactory]
+        elif task == FactoryTypes.VISION:
+            # vision
+            from vision.factory import CompressionFactory as VisionFactory
+
+            cls = [VisionFactory]
+    elif job_service == JobServices.ADAPT:
+        task: AdaptTasks = AdaptTasks(
+            task,
+        )
```

Comment:
1. This was black formatted. 
2. This variable is created now incase the code changes in the future and we decide to use this variable instead of using strings to define tasks. But currently that would have required additional code changes for something trivial.

Reply:
if need be, the variable will be created then. I dont see a reason to have unused hanging variables or imports once noticed.

Reply:
done.